server/src/test/java/io/crate/planner/ExplainPlannerTest.java

```diff
@@ -121,10 +167,48 @@ public void test_explain_on_collect_uses_cast_optimizer_for_query_symbol() throw
             .addTable("CREATE TABLE ts1 (ts TIMESTAMP)")
             .build();
 
-        ExplainPlan plan = e.plan("EXPLAIN SELECT * FROM ts1 WHERE ts = 1662740986992");
-        var printedPlan = ExplainPlan.printLogicalPlan((LogicalPlan) plan.subPlan(), e.getPlannerContext(clusterService.state()));
+        ExplainPlan plan = e.plan("EXPLAIN (COSTS FALSE) SELECT * FROM ts1 WHERE ts = 1662740986992");
+        var printedPlan = ExplainPlan.printLogicalPlan((LogicalPlan) plan.subPlan(), e.getPlannerContext(clusterService.state()), plan.showCosts());
         assertThat(printedPlan).isEqualTo(
             "Collect[doc.ts1 | [ts] | (ts = _cast(1662740986992::bigint, 'timestamp without time zone'))]"
         );
     }
+
+    @Test
+    public void test_explain_costs_adds_estimated_rows_to_output() throws Exception {
+        var e = SQLExecutor.builder(clusterService)
+            .addTable("CREATE TABLE doc.a (x int)")
+            .addTable("CREATE TABLE doc.b (x int)")
+            .build();
+
+        e.updateTableStats(Map.of(
+            new RelationName("doc", "a"), new Stats(100, 100, Map.of()),
+            new RelationName("doc", "b"), new Stats(100, 100, Map.of())
+        ));
+
+        ExplainPlan plan = e.plan("EXPLAIN SELECT COUNT(a.x) FROM a join b on a.x = b.x");
+        var printedPlan = ExplainPlan.printLogicalPlan((LogicalPlan) plan.subPlan(), e.getPlannerContext(clusterService.state()), plan.showCosts());
+        assertThat(printedPlan).isEqualTo(
+            "HashAggregate[count(x)] (rows=1)\n" +
+            "  └ HashJoin[(x = x)] (rows=0)\n" +
```

Comment:
minor: could we add comments about how the `rows` came out to be `1` and `0`?

Reply:
This is related to the cardinality estimation, the table is empty and nothing matches.